kotsadm/pkg/apiserver/server.go

```diff
@@ -161,3 +167,68 @@ func Start() {
 
 	log.Fatal(srv.ListenAndServe())
 }
+
+// Detects the InstanceID of kodsadm pod across restores
+func detectInstanceID() {
+
+	log.Println("Detecting Instance")
+
+	// Retrieve the ClusterID from store
+	clusters, err := store.GetStore().ListClusters()
+	if err != nil {
+		logger.Errorf("Failed to list clusters %v", err)
+		return
+	}
+	if len(clusters) == 0 {
+		logger.Error(errors.New("No clusters found"))
+		return
+	}
+	clusterID := clusters[0].ClusterID
+
+	// Write a Query to set/get an Event from the Store
+	keyExists, err := store.GetStore().GetKotsAdmEventStatus()
```

Comment:
i would also suggest renaming this to something like
```suggestion
	keyExists, err := store.GetStore().IsKotsadmIDGenerated()
```